src/contract.ts

```diff
@@ -1,376 +1,479 @@
-      : program.sendAndConfirmTransaction
-    ).call(program, [instruction], signers);
-
-    const contract = new Contract(program, storageAccount, contractAbiData);
-
-    const events = contract.parseLogsEvents(logs);
-
-    return {
-      contract,
-      logs,
-      computeUnitsUsed,
-      events,
-    };
-  }
-
-  /**
-   * Load a deployed contract
-   *
-   * @param program
-   * @param abiData
-   * @param storageAccount
-   * @returns
-   */
-  static async get(
-    program: Program,
-    abiData: string,
-    storageAccount: PublicKey
-  ): Promise<Contract> {
-    return new Contract(program, storageAccount, abiData);
-  }
-
-  public abi: Interface;
-  readonly functions: { [name: string]: ContractFunction };
-
-  /**
-   * Creates a new instance of Contract
-   *
-   * @param program
-   * @param storageAccount
-   * @param abiData
-   */
-  constructor(_program: Program, _storageAccount: PublicKey, _abiData: string) {
-    this.program = _program;
-    this.storageAccount = _storageAccount;
-    this.abiData = _abiData;
-    this.abi = new Interface(_abiData);
-    this.functions = {};
-    Object.values(this.abi.functions).forEach((frag) => {
-      this.functions[frag.name] = this.buildCall(frag);
-    });
-  }
-
-  /**
-   * Generate contract method for `fragment` of type "function"
-   *
-   * @param fragment
-   * @returns
-   */
-  buildCall(fragment: Fragment): ContractFunction {
-    return (...args: Array<any>) => {
-      const last = args[args.length - 1];
-      if (args.length > fragment.inputs.length && typeof last === 'object') {
-        return this.call(
-          fragment.name,
-          args.slice(0, fragment.inputs.length),
-          last
+    /** @TODO: docs */
+    readonly [name: string]: ContractFunction | any;
+
+    /** Connection to use */
+    readonly connection: Connection;
+    /** Account the program is located at (aka Program ID) */
+    readonly program: PublicKey;
+    /** Account the program's data is stored at */
+    readonly storage: PublicKey;
+    /** Application Binary Interface in JSON form */
+    readonly abi: Abi;
+    /** Ethers.js interface parsed from the ABI */
+    readonly interface: Interface;
+    /** Callable functions mapped to the interface */
+    readonly functions: Record<string, ContractFunction>;
+    /** Parser for events and program logs */
+    readonly logs: LogsParser;
+    /** Payer for transactions and storage (optional) */
+    payer: Signer | null;
+
+    /*
+     * Create a contract. It can either be a new contract to deploy as a Solana program,
+     * or a reference to one already deployed.
+     *
+     * @param connection Connection to use
+     * @param program    Account the program is located at (aka Program ID)
+     * @param storage    Account the program's data is stored at
+     * @param abi        Application Binary Interface in JSON form
+     * @param payer      Payer for transactions and storage (optional)
```

Comment:
Currently `payer` isn't really optional, because to simulate transactions to read from accessors, a signer is required, and will be checked to have enough lamports to cover signature verification.

But it doesn't necessarily need to be provided on construction, and the `connect` method similar to that in Ethers.js can be used to add a payer.

This requirement may be relaxed once storage can be deserialized directly without simulation.